src/main/java/uk/gov/companieshouse/documentsigningapi/interceptor/UserAuthenticationInterceptor.java

```diff
@@ -0,0 +1,51 @@
+package uk.gov.companieshouse.documentsigningapi.interceptor;
+
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
+import uk.gov.companieshouse.documentsigningapi.logging.LoggingUtilsConfiguration;
+import uk.gov.companieshouse.documentsigningapi.util.EricHeaderHelper;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.springframework.http.HttpStatus.UNAUTHORIZED;
+import static uk.gov.companieshouse.documentsigningapi.logging.LoggingUtilsConfiguration.*;
+
+@Component
+public class UserAuthenticationInterceptor implements HandlerInterceptor {
+    private static final Logger LOGGER = LoggerFactory.getLogger(LoggingUtilsConfiguration.APPLICATION_NAMESPACE);
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) {
+        Map<String, Object> logMap = new HashMap<>();
+        logMap.put(REQUEST_ID_LOG_KEY, request.getHeader(REQUEST_ID_HEADER_NAME));
+        //
+        // Any value will do for ERIC-Identity-Type so long as it is NOT empty.
+        //
+        final String identityType = EricHeaderHelper.getIdentityType(request);
```

Comment:
I know we often end up with a mixture of `final` and non-`final` variables as our code evolves with different styles, but with new code we can be consistent in either using `final` where we know something should not change, or just not bother with using `final` at all (except in "constants") if we don't believe it adds value? 

Not a big deal.